crates/starknet-devnet-server/src/api/json_rpc/mod.rs

```diff
@@ -109,6 +110,52 @@ impl RpcHandler for JsonRpcHandler {
         info!(target: "rpc", "received method in on_request {}", request);
         self.execute(request, original_call).await
     }
+
+    async fn on_call(&mut self, call: RpcMethodCall) -> RpcResponse {
+        trace!(target: "rpc",  id = ?call.id , method = ?call.method, "received method call");
+        let RpcMethodCall { method, params, id, .. } = call.clone();
+
+        let params: serde_json::Value = params.into();
+        let deserializable_call = serde_json::json!({
+            "method": &method,
+            "params": params
+        });
+
+        match serde_json::from_value::<Self::Request>(deserializable_call) {
+            Ok(req) => {
+                if let Some(restricted_methods) = &self.server_config.restricted_methods {
+                    if is_json_rpc_method_restricted(
+                        &method,
+                        restricted_methods
+                            .iter()
+                            .map(|x| x.as_str())
+                            .collect::<Vec<&str>>()
+                            .as_slice(),
```

Comment:
Here and in the HTTP middleware, we will be doing this conversion on every call? Can't it be simplified, e.g. so that the statics hold `String` instead of `&str`?

Reply:
https://github.com/0xSpaceShard/starknet-devnet-rs/commit/edf343ce913446fe388c305d2c6d5ea0c668118a